cmd/rpcdaemon/cli/config.go

```diff
@@ -504,12 +505,12 @@ func RemoteServices(ctx context.Context, cfg *httpcfg.HttpCfg, logger log.Logger
 		if cc != nil && cc.Bor != nil {
 			if polygonSync {
 				stateReceiverContractAddress := cc.Bor.GetStateReceiverContract()
-				bridgeReader, err = bridge.AssembleReader(ctx, cfg.DataDir, logger, stateReceiverContractAddress)
+				bridgeReader, err = bridge.AssembleReader(ctx, cfg.DataDir, logger, stateReceiverContractAddress, roTxLimit)
 				if err != nil {
 					return nil, nil, nil, nil, nil, nil, nil, ff, nil, nil, err
 				}
 
-				heimdallReader, err = heimdall.AssembleReader(ctx, cc.Bor.CalculateSprintNumber, cfg.DataDir, cfg.Dirs.Tmp, logger)
+				heimdallReader, err = heimdall.AssembleReader(ctx, cc.Bor.CalculateSprintNumber, cfg.DataDir, cfg.Dirs.Tmp, logger, roTxLimit)
```

Comment:
Would it be possible to combine ` cfg.DataDir, cfg.Dirs.Tmp, logger ` and `roTxLimit` to a `DBConfig` struct to avoid adding yet another parameter?

Reply:
yes, I agree with Elton, I actually raised the same thing in a previous PR - https://github.com/erigontech/erigon/pull/11924#discussion_r1750056992

Reply:
I even suggest we have separate config structs per component like
- `heimdall.ReaderConfig` and `bridge.ReaderConfig`
- `heimdall.ServiceConfig` and `bridge.ServiceConfig`
- etc.

Reply:
to keep this productive though, maybe this PR can be merged and these improvements can be done in a follow up?

Reply:
ok, I'm approving and leaving it up to Shoham.